internal/codec/codec_test.go

```diff
@@ -0,0 +1,20 @@
+package codec
+
+import (
+	"fmt"
+	"os"
+)
+
+func Example() {
+	enc := GetEncoder(os.Stdout)
+	defer PutEncoder(enc)
+	enc.MustEncode([]string{"a", "slice", "of", "strings"})
+	fmt.Fprintln(os.Stdout)
+	enc.MustEncode(nil)
+	fmt.Fprintln(os.Stdout)
+	enc.MustEncode(map[string]string{})
+	fmt.Fprintln(os.Stdout)
+	// Output: ["a","slice","of","strings"]
```

Comment:
probably want to test with json large then your 4096 buffer size to make sure things are OK.